evennia/contrib/actions/README.md

```diff
@@ -0,0 +1,372 @@
+
+The movement speed (movespeed) function has a single argument, a reference to the character object itself. It must return a floating-point value in distance units per second. Distance units are the units of measurement assigned to exits to determine how far one must travel to get from the middle of the room to the exit, or conversely from the exit to the middle of the room. At the bare minimum, a simple movespeed function can simply return a floating-point value that represents the character's permanent movement speed. More complex movespeed functions can invoke the character object's action.movetype attribute and pick a given movement speed based on the movement type being used, as well as other factors such as the character's agility.
+
+To have characters set their movement type during the game, such as through a "pace" command, ensure that the command does the following:
+
+>     <character object>.actions.movetype = <your string here>
+
+It is not necessary to include either movespeed or movetype. When you fail to include movespeed or movetype, the movement speed of the character will default to 1 distance unit per second. If, furthermore, the exit that the character chooses to traverse has a distance value of 1, then it will take 1 second to traverse it. Thus, you usually want to set the exits' distance values or give the characters movespeed functions, or both.
+
+The movebps function also has a single argument, a string representing the movement type being performed. You will want this string to be one you would assign to self.actions.movetype. The movebps function returns a string of comma-separated bodypart names, the same as you would find in an action's "bodyparts" item. While it is not necessary to set this function, the consequence of not doing so is that all movement actions will use no bodyparts at all, meaning that characters could issue a vast number of concurrent movement actions, as well as start moving to an exit immediately upon entering a room, rather than as soon as their MoveIn action was completed.
+
+When you subclass from __ActionRoom__, you should do: 
+
+> at_object_creation(self)
+>     self.actions.view = <your function here>
+>    *[...]*
+>     super(ActionRoom, self).at_object_creation()
+
+The view function must have two arguments. The first of these is the object that is being viewed and the second is the character viewing it. It will return either a formatted string or the value False, the former when the object being viewed can be seen by the viewer, and the latter when the object being viewed is invisible to the viewer. The string can contain the name of the object being viewed, its description or anything else the coder desires. The view function is used automatically inside the ActionRoom's "actions.display" method and should never need to be caled directly.
+
+Finally, when you subclass from __ActionExit__, you should do:
+> at_object_creation(self)
+>     self.distance = <your float here>
```

Comment:
I see. This should probably be clarified since it's can be (and is, as you notice) confusing compared to other Evennia APIs. 

Standard Evennia tries to be clear with what is persistent or not - it has four cases: 

1. .db for persistent Attributes, 
2. .ndb for non-persistent Attributes.
3. Handlers, which are called as `obj.handlername.add()/remove()` etc (never on their own). 
4. Model-centric database fields which are few and non-changing, like `self.key` and `self.location`.
5. All other properties are non-persistent. 

By introducing a new type of persistent property you are introducing a sixth case. If done for a good reason this is not wrong per se, but it's different from Evennia standard API and should be pointed out.